opm/autodiff/SimulatorFullyImplicitBlackoilOutput.hpp

```diff
@@ -375,5 +378,179 @@ namespace Opm
     }
 
 
+
+
+
+    namespace detail {
+
+        struct WriterCall : public ThreadHandle :: ObjectInterface
+        {
+            BlackoilOutputWriter& writer_;
+            std::unique_ptr< SimulatorTimerInterface > timer_;
+            const SimulationDataContainer state_;
+            const WellState wellState_;
+            std::vector<data::CellData> simProps_;
+            const bool substep_;
+
+            explicit WriterCall( BlackoilOutputWriter& writer,
+                                 const SimulatorTimerInterface& timer,
+                                 const SimulationDataContainer& state,
+                                 const WellState& wellState,
+                                 const std::vector<data::CellData>& simProps,
+                                 bool substep )
+                : writer_( writer ),
+                  timer_( timer.clone() ),
+                  state_( state ),
+                  wellState_( wellState ),
+                  simProps_( simProps ),
+                  substep_( substep )
+            {
+            }
+
+            // callback to writer's serial writeTimeStep method
+            void run ()
+            {
+                // write data
+                writer_.writeTimeStepSerial( *timer_, state_, wellState_, simProps_, substep_ );
+            }
+        };
+
+
+
+
+        template<class Model>
+        std::vector<data::CellData> getCellData(const Model& model,
+                const RestartConfig& restartConfig,
+                const int reportStepNum) {
+            std::vector<data::CellData> simProps;
+
+
+            std::map<const char*, int> outKeywords {
+                {"ALLPROPS", 0},
```

Comment:
This is related to your comment below (expansion of the ALLPROPS keyword). I'll double check what's going on on my side.